jupyter_server/serverapp.py

```diff
@@ -1134,13 +1136,43 @@ def template_file_path(self):
         help="""If True, display controls to shut down the Jupyter server, such as menu items or buttons."""
     )
 
-    contents_manager_class = Type(
+    # REMOVE in VERSION 2.0
+    # Temporarily allow content managers to inherit from the 'notebook'
+    # package. We will deprecate this in the next major release.
+    contents_manager_class = TypeFromClasses(
         default_value=LargeFileManager,
-        klass=ContentsManager,
+        klasses=[
+            'jupyter_server.services.contents.manager.ContentsManager',
+            'notebook.services.contents.manager.ContentsManager'
+        ],
         config=True,
         help=_('The content manager class to use.')
     )
 
+    # Throws a deprecation warning to notebook based contents managers.
+    @observe('contents_manager_class')
+    def _observe_contents_manager_class(self, change):
+        new = change['new']
+        # If 'new' is a class, get a string representing the import
+        # module path.
+        if inspect.isclass(new):
+            new = new.__module__
+
+        if new.startswith('notebook'):
+            self.log.warn(
```

Comment:
I believe `warn()` will give a deprecation warning so we should probably use `warning()` here.